packages/react/src/components/Tabs/tabs.test.tsx

```diff
@@ -0,0 +1,411 @@
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  expect(screen.getByRole('tablist')).toHaveAttribute('aria-label', 'find-me');
+});
+
+test('should render with aria-labelledby prop', () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    return (
+      <>
+        <Tabs aria-labelledby="find-me">
+          <Tab target={ref}>option 1</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  expect(screen.getByRole('tablist')).toHaveAttribute(
+    'aria-labelledby',
+    'find-me'
+  );
+});
+
+test('should display correct tab panel when clicking a tab', async () => {
+  const TabswithRef = () => {
+    const ref = useRef(null);
+    const ref2 = useRef(null);
+    const ref3 = useRef(null);
+    return (
+      <>
+        <Tabs aria-label={ariaLabel}>
+          <Tab target={ref}>option 1</Tab>
+          <Tab target={ref2}>option 2</Tab>
+          <Tab target={ref3}>option 3</Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+        <TabPanel ref={ref2}>
+          <p>Panel 2</p>
+        </TabPanel>
+        <TabPanel ref={ref3}>
+          <p>Panel 3</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabswithRef />);
+  fireEvent.click(screen.getAllByRole('tab')[0]);
+  await waitFor(() => {
+    expect(screen.getAllByRole('tabpanel')[0]).not.toHaveClass(
+      'TabPanel--hidden'
+    );
+    expect(screen.getAllByRole('tabpanel')[1]).toHaveClass('TabPanel--hidden');
+    expect(screen.getAllByRole('tabpanel')[2]).toHaveClass('TabPanel--hidden');
+  });
+
+  fireEvent.click(screen.getAllByRole('tab')[1]);
+  await waitFor(() => {
+    expect(screen.getAllByRole('tabpanel')[0]).toHaveClass('TabPanel--hidden');
+    expect(screen.getAllByRole('tabpanel')[1]).not.toHaveClass(
+      'TabPanel--hidden'
+    );
+    expect(screen.getAllByRole('tabpanel')[2]).toHaveClass('TabPanel--hidden');
+  });
+});
+
+test('should focus the active tab with keyboard activation', async () => {
+  const mountElement = document.createElement('div');
+  document.body.appendChild(mountElement);
+  const TabsComponent = () => {
+    const ref = useRef(null);
+    const ref2 = useRef(null);
+
+    return (
+      <>
+        <Tabs aria-label={ariaLabel}>
+          <Tab id="tab-uno" target={ref}>
+            option 1
+          </Tab>
+          <Tab id="tab-dos" target={ref2}>
+            option 2
+          </Tab>
+        </Tabs>
+        <TabPanel ref={ref}>
+          <p>Panel 1</p>
+        </TabPanel>
+        <TabPanel ref={ref2}>
+          <p>Panel 2</p>
+        </TabPanel>
+      </>
+    );
+  };
+
+  render(<TabsComponent />, { container: mountElement });
+  fireEvent.keyDown(screen.getByText('option 2'), { key: 'ArrowRight' });
```

Comment:
According to [the RTL docs](https://testing-library.com/docs/guide-events/#keydown), I believe it's best practice to focus the element first then fire the key down event on the active element. 
```suggestion
  screen.getByText('option 2').focus()
  fireEvent.keyDown(document.activeElement || document.body, { key: 'ArrowRight' });
```